app/scripts/controllers/network/create-network-client.ts

```diff
@@ -57,7 +57,11 @@ export function createNetworkClient(
   const rpcApiMiddleware =
     networkConfig.type === NetworkClientType.Infura
       ? createInfuraMiddleware({
-          network: networkConfig.network,
+          network:
+            // TODO: this condition is to test the linea mainnet implemenation because the mainnet infrastructure is not yet ready. TO BE REMOVED
+            networkConfig.network === 'linea-mainnet'
```

Comment:
Is this still needed? If so, when is this planned to be removed?

Reply:
Yes it is needed because we use the testnet RPC url for now. We will update this url once the linea mainnet network is ready.

Reply:
Hiding the Linea mainnet network prior to release makes sense, but I don't see the purpose in making it act like Linea Goerli. If we do this, then it will require another code change to make the network work properly. If that's so, then why bother adding it all until that point?

Reply:
Indeed :) The idea was to put the linea-goerli rpc url for the linea mainnet to test the integration but I think it is better to put the real rpc url. I will change that thanks :)